scrapscript.py

```diff
@@ -2122,6 +2185,15 @@ def test_match_with_inequal_ints_returns_none(self) -> None:
     def test_match_int_with_non_int_returns_none(self) -> None:
         self.assertEqual(match(String("abc"), pattern=Int(1)), None)
 
+    def test_match_with_equal_floats_returns_empty_dict(self) -> None:
```

Comment:
I don't know that we should support direct match on floats. Float equality can be very finicky. Perhaps we should have a way to do with some epsilon

Reply:
Probably worth doing whatever other languages with pattern matching do if it's easy. I could imagine some languages might just allow you to do it and you should know that sometimes floats will be not quite equal and you won't get a match. If it's not easy I'll just remove pattern matching for floats here.

Reply:
In Rust matching on floats is a warning that will (soon?) be turned into an error. 
https://github.com/rust-lang/rust/issues/41255
https://github.com/rust-lang/rust/issues/

In Haskell, Dart, and some others matching on floats seems supported.

I think it's fair to just not support pattern matching on floats for now since it's not so pressing

Reply:
Removed